static/app/views/performance/transactionSummary/transactionSpans/spanSummary/content.spec.tsx

```diff
@@ -0,0 +1,304 @@
+              },
+            ],
+          ],
+          [
+            1717106400,
+            [
+              {
+                count: 143062.61666666667,
+              },
+            ],
+          ],
+          [
+            1717110000,
+            [
+              {
+                count: 158031.58333333334,
+              },
+            ],
+          ],
+        ],
+      },
+    });
+
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/events/',
+      body: {
+        data: [
+          {
+            'transaction.id': '93b88037ba134225bdf67d05a69de9ab',
+            project: 'sentry',
+            'project.name': 'sentry',
+            span_id: '9b6e1f295ce7e875',
+            id: '9b6e1f295ce7e875',
+            'span.duration': 0.860929,
+            trace: '80a8718f4b3847eb8d6f3b5715602558',
+            timestamp: '2024-05-16T14:45:15+00:00',
+          },
+        ],
+      },
+    });
+
+    MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/events/',
+      body: {
+        data: [
+          {
+            'transaction.duration': 160,
+            id: '93b88037ba134225bdf67d05a69de9ab',
+            'project.name': 'sentry',
+          },
+          {
+            'transaction.duration': 50,
+            id: '2a2c0e1a7cf941f6bcd8ab22b0c4d8c9',
+            'project.name': 'sentry',
+          },
+        ],
+      },
+    });
+  });
+
+  it('correctly renders the details in the header', async function () {
+    headerDataMock = MockApiClient.addMockResponse({
+      url: '/organizations/org-slug/events/',
+      body: {
+        data: [
+          {
+            'span.description': 'SELECT thing FROM my_cool_db WHERE value = %s',
+            'avg(span.duration)': 1.7381229881349218,
+            'count()': 3677407172,
+            'sum(span.self_time)': 6391491809.035965,
+          },
+        ],
+        meta: {
+          fields: {
+            'span.description': 'string',
+            'sum(span.self_time)': 'duration',
+            'count()': 'integer',
+            'avg(span.duration)': 'duration',
+          },
+          units: {
+            'span.description': null,
+            'sum(span.self_time)': 'millisecond',
+            'count()': null,
+            'avg(span.duration)': 'millisecond',
+          },
+          isMetricsData: false,
+          isMetricsExtractedData: false,
+          tips: {},
+          datasetReason: 'unchanged',
+          dataset: 'spansMetrics',
+        },
+      },
+    });
+
+    render(
+      <SpanSummary
+        spanSlug={{group: 'aaaaaaaa', op: 'db'}}
+        transactionName="transaction"
+        organization={organization}
+        project={undefined}
+      />
+    );
+
+    expect(headerDataMock).toHaveBeenCalled();
+
+    const headerContainerOp = await screen.findByTestId('operation-name');
+    const headerContainerDescription = await screen.findByTestId(
+      'header-span-description'
+    );
```

Comment:
If this is a header, might be better to do something like `screen.getByRole('header', {level: 2})` and check its contents (sorry, I forgot the exact syntax, but the idea is that querying by role is more flexible)

Reply:
It's built with just divs so I can't use this unfortunately, but I'll keep that in mind when writing future tests!